src/js/wall/FirstUser.tsx

```diff
@@ -0,0 +1,95 @@
+import { useMutation } from "@tanstack/react-query";
+import React from "react";
+import { useForm } from "react-hook-form";
+import { InputError, InputGroup, InputSimple } from "../base";
+import { createFirst } from "../users/api";
+import { User } from "../users/types";
+import { WallButton, WallContainer, WallDialog, WallHeader, WallLoginContainer, WallSubheader } from "./Container";
+import { WallTitle } from "./WallTitle";
+
+type ErrorResponse = {
+    response: {
+        body: {
+            message: string;
+        };
+    };
+};
+
+type NewUser = {
+    handle: string;
+    password: string;
+    /** Whether the user will be forced to reset their password on next login */
+    forceReset: boolean;
+};
+
+type FormData = {
+    username: string;
+    password: string;
+};
+
+/**
+ * A form for creating the first instance user
+ */
+export default function FirstUser() {
+    const mutation = useMutation<User, ErrorResponse, NewUser>(createFirst, {
+        onSuccess: () => {
+            window.location.reload();
+        },
+    });
+
+    const {
+        formState: { errors },
+        handleSubmit,
+        register,
+    } = useForm<FormData>();
+
+    function onSubmit(data: FormData) {
+        mutation.mutate({
+            handle: data.username,
+            password: data.password,
+            forceReset: false,
+        });
+    }
+
+    return (
+        <WallContainer>
+            <WallDialog>
+                <WallLoginContainer>
+                    <WallTitle />
+                    <WallHeader>Setup First User</WallHeader>
+                    <WallSubheader>
+                        Create an administrative user that can be used to configure your new Virtool instance.
+                    </WallSubheader>
+
+                    <form onSubmit={handleSubmit(onSubmit)}>
+                        <InputGroup>
+                            <InputSimple aria-label="username" id="username" type="username" defaultValue="" />
```

Comment:
The `InputSimple` for the username doesn't register anything so you may want to double check on that one. I believe you can remove the `type`. You should also add the `InputLabel` back as it gives context for the users. To access the input in test you can just add `htmlFor` on `InputLabel`

Also you should add error handling for the register.

Reply:
I added the `InputLabel` back and updated the test, but I'm not quite sure how to add error handling for the register. Could you point me in the right direction for that?

Reply:
Yeah, I just added some comments to see if that alternative would work instead